orangecontrib/geo/widgets/owmap.py

```diff
@@ -1,906 +1,849 @@
-        QTimer.singleShot(0, _toggle_legend)
-        QTimer.singleShot(0, _set_opacity)
-        QTimer.singleShot(0, _set_zoom)
-        QTimer.singleShot(0, _set_jittering)
-        QTimer.singleShot(0, _set_clustering)
-
-    autocommit = settings.Setting(True)
-
-    def onDeleteWidget(self):
-        self.map.stop = True
-        super().onDeleteWidget()
-
-    def commit(self):
-        self.Outputs.selected_data.send(self.selection)
-        self.Outputs.annotated_data.send(create_annotated_table(self.data, self._indices))
-
-    @Inputs.data
     def set_data(self, data):
-        self.data = data
-
-        self.closeContext()
-
-        if data is None or not len(data):
-            return self.clear()
-
-        domain = data is not None and data.domain
-        for model in (self._latlon_model,
-                      self._class_model,
-                      self._color_model,
-                      self._shape_model,
-                      self._size_model,
-                      self._label_model):
-            model.set_domain(domain)
-
-        self.lat_attr, self.lon_attr = find_lat_lon(data)
-        if data.domain.class_var:
-            self.color_attr = data.domain.class_var
-
-        self.openContext(data)
-
-        self.map.set_data(self.data, self.lat_attr, self.lon_attr, redraw=False)
-        self.map.set_marker_color(self.color_attr, update=False)
-        self.map.set_marker_label(self.label_attr, update=False)
-        self.map.set_marker_shape(self.shape_attr, update=False)
-        self.map.set_marker_size(self.size_attr, update=True)
-
-    @Inputs.data_subset
-    def set_subset(self, subset):
-        self.map.set_subset_ids(subset.ids if subset is not None else np.array([]))
-
-    def handleNewSignals(self):
-        super().handleNewSignals()
-        self.train_model()
-
-    @Inputs.learner
-    def set_learner(self, learner):
-        self.learner = learner
-        self.controls.class_attr.setEnabled(learner is not None)
-        self.controls.class_attr.setToolTip(
-            'Needs a Learner input for modelling.' if learner is None else '')
-
-    def train_model(self):
-        model = None
-        self.Error.clear()
-        if self.data and self.learner and self.class_attr != '(None)':
-            domain = self.data.domain
-            if self.lat_attr and self.lon_attr and self.class_attr in domain:
-                domain = Domain([domain[self.lat_attr], domain[self.lon_attr]],
-                                [domain[self.class_attr]])  # I am retarded
-                train = Table.from_table(domain, self.data)
-                try:
-                    model = self.learner(train)
-                except Exception as e:
-                    self.Error.learner_error(e)
-        self.map.set_model(model)
-
-    def disable_some_controls(self, disabled):
-        tooltip = (
-            "Available when the zoom is close enough to have "
-            "<{} points in the viewport.".f
```

Comment:
`Map,` `Symbol size`, `opacity`, `jittering` and `show legend` settings are not properly migrated.